pkgs/tools/package-management/nix-serve/default.nix

```diff
@@ -12,23 +21,34 @@ stdenv.mkDerivation rec {
     sha256 = "0mjzsiknln3isdri9004wwjjjpak5fj8ncizyncf5jv7g4m4q1pj";
   };
 
-  buildInputs = [ perl nix ]
-    ++ (with perlPackages; [ DBI DBDSQLite Plack Starman ]);
+  buildInputs = [ makeWrapper ];
 
-  dontBuild = false;
+  propagatedBuildInputs = [
+    nix
+    perl
+    perlPackages.DBDSQLite
+    perlPackages.DBI
+    perlPackages.Plack
+    perlPackages.Starman
+  ];
+
+  phases = ["unpackPhase" "installPhase"];
 
-  # FIXME: unfortunate cut&paste.
   installPhase = ''
     mkdir -p $out/libexec/nix-serve
     cp nix-serve.psgi $out/libexec/nix-serve/nix-serve.psgi
 
     mkdir -p $out/bin
     cat > $out/bin/nix-serve <<EOF
     #! ${stdenv.shell}
-    PERL5LIB=$PERL5LIB exec ${perlPackages.Starman}/bin/starman $out/libexec/nix-serve/nix-serve.psgi "\$@"
+    exec ${perlPackages.Starman}/bin/starman $out/libexec/nix-serve/nix-serve.psgi "\$@"
     EOF
     chmod +x $out/bin/nix-serve
-  '';
+
+    wrapProgram $out/bin/nix-serve \
```

Comment:
I considered that, and I'm not opposed to it per se. On the other hand, the `wrapProgram` script is used all over `nixpkgs`. But I'm fine either way.